File: Leetcode_1768.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        a = list(word1)
        b = list(word2)
        result = ""

        min_len = min(len(a), len(b))
        for i in range(min_len):
            result += a[i] + b[i]
        result += ''.join(a[min_len:]) + ''.join(b[min_len:])

        return result

File: Leetcode_1768.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        char[] a = word1.toCharArray();
        char[] b = word2.toCharArray();
        StringBuilder result = new StringBuilder();

        int minLen = Math.min(a.length, b.length);

        for (int i = 0; i < minLen; i++) {
            result.append(a[i]).append(b[i]);
        }
        if (a.length > minLen) {
            result.append(word1.substring(minLen));
        }

        if (b.length > minLen) {
            result.append(word2.substring(minLen));
        }

        return result.toString();
    }
}
